crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -18,236 +18,234 @@
+        cluster_health = {}
+        crm_failures = {}
+        failed_actions = {}
+
+        founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
+        return true if founders.empty?
+
+        founders.each do |n|
+          name = n.name
+          ssh_retval = n.run_ssh_cmd("crm status 2>&1")
+          if ssh_retval[:exit_code] != 0
+            crm_failures[name] = ssh_retval[:stdout]
+            Rails.logger.warn(
+              "crm status at node #{name} reports error:\n#{ssh_retval[:stdout]}"
+            )
+            next
+          end
+          ssh_retval = n.run_ssh_cmd("LANG=C crm status | grep -A 2 '^Failed Actions:'")
+          if ssh_retval[:exit_code] == 0
+            failed_actions[name] = ssh_retval[:stdout]
+            Rails.logger.warn(
+              "crm at node #{name} reports some failed actions:\n#{ssh_retval[:stdout]}"
+            )
+          end
+        end
+        cluster_health["crm_failures"] = crm_failures unless crm_failures.empty?
+        cluster_health["failed_actions"] = failed_actions unless failed_actions.empty?
 
-        os_available = repo_version_available?(products, "SLES", "12.2")
-        ret[:os] = {
-          available: os_available,
-          repos: {}
-        }
-        ret[:os][:repos][admin_architecture.to_sym] = {
-          missing: ["SUSE Linux Enterprise Server 12 SP2"]
-        } unless os_available
-
-        cloud_available = repo_version_available?(products, "suse-openstack-cloud", "7")
-        ret[:cloud] = {
-          available: cloud_available,
-          repos: {}
-        }
-        ret[:cloud][:repos][admin_architecture.to_sym] = {
-          missing: ["SUSE OpenStack Cloud 7"]
-        } unless cloud_available
+        cluster_health.empty?
       end
-    end
-
 
-    # Simple check if HA clusters report some problems
-    # If there are no problems, empty hash is returned.
-    # If this fails, information about failed actions for each cluster founder is
-    # returned in a hash that looks like this:
-    # {
-    #     "crm_failures" => {
-    #             "node1" => "reason for crm status failure"
-    #     },
-    #     "failed_actions" => {
-    #             "node2" => "Failed action on this node"
-    #     }
-    # }
-    # User has to manually clean pacemaker resources before proceeding with the upgrade.
-    def clusters_healthy?
-      cluster_health = {}
-      crm_failures = {}
-      failed_actions = {}
-
-      founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
-      return true if founders.empty?
-
-      founders.each do |n|
-        name = n.name
-        ssh_retval = n.run_ssh_cmd("crm status 2>&1")
-        if ssh_retval[:exit_code] != 0
-          crm_failures[name] = ssh_retval[:stdout]
-          Rails.logger.warn(
-            "crm status at node #{name} reports error:\n#{ssh_retval[:stdout]}"
-          )
-          next
+      def ceph_healthy?
```

Comment:
Looks like you reordered some methods, e.g. `ceph_healthy?` got moved before `compute_resources_available?`.  That's absolutely fine (I'm a fan of putting methods in a logical order), but it's a refactoring which should happen in a separate commit saying "reorder methods with no other changes", so that reviewers know (and can easily check if they want) that it is simply a method reordering, with no functional changes.  When it's mixed with other changes, it makes the review much harder, because each change can't be viewed atomically, and then it's easier to miss something in the review.

I know this is just a cherry-pick - sorry for not mentioning this earlier in the original review in #674. But I hope it's still worth mentioning here as something to bear in mind for the future.